### PR TITLE
AuthN: Login error handling

### PR DESCRIPTION
**Pull request by kalleep**
**What is this feature?**
Add missing public messages for some oauth errors and create errutil.Error from social.Error.
Also Change from rendering error page to use redirect with error cookie

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/login/social/common.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ import (
 )
 
 var (
-	errMissingGroupMembership = Error{"user not a member of one of the required groups"}
+	errMissingGroupMembership = &Error{"user not a member of one of the required groups"}
```
Comment (kalleep) [inline pkg/login/social/common.go:15]: This will now make this error show up as a public error

File: pkg/api/login_oauth.go [hunk 2]
```diff
@@ -153,7 +153,7 @@ func (hs *HTTPServer) OAuthLogin(ctx *contextmodel.ReqContext) {
 
 		state, err := GenStateString()
 		if err != nil {
-			ctx.Logger.Error("Generating state string failed", "err", err)
+			ctx.Logger.Error("Generating state stringit push --set-upstream origin kalleep/authn/login-error-handlingg failed", "err", err)
```
Comment (Jguer) [inline pkg/api/login_oauth.go:156]: mid button press I think :)

Comment (Jguer) [inline pkg/api/login_oauth.go:156]:
```suggestion
			ctx.Logger.Error("Generating state string failed", "err", err)
```

**Review by Jguer — APPROVED** [review]
Good after changing typo

File: pkg/api/login_oauth.go (hunk 2 above)
Comment (kalleep) [reply pkg/api/login_oauth.go]: ops

**Review by eleijonmarck — APPROVED** [review]

**Pull request merged**